Mage.Tests/src/test/java/org/mage/test/cards/single/ncc/ThreefoldSignalTest.java

```diff
@@ -0,0 +1,104 @@
+package org.mage.test.cards.single.ncc;
+
+import mage.constants.PhaseStep;
+import mage.constants.Zone;
+import org.junit.Test;
+import org.mage.test.serverside.base.CardTestPlayerBase;
+
+/**
+ * {@link mage.cards.t.ThreefoldSignal Threefold Signal}
+ * <p>
+ * Each spell you cast that’s exactly three colors has replicate {3}.
+ * @author Alex-Vasile
+ */
+public class ThreefoldSignalTest extends CardTestPlayerBase {
```

Comment:
you should add a test for split cards (including fuse) and MDFCs

Reply:
Will do. Though there are no 3 color MDFCs for me to test this with.

Reply:
there’s [three](https://scryfall.com/search?q=is%3Amdfc+ci%3E2&unique=cards&as=grid&order=name) you can use

Reply:
Is this card looking at color identity or color? I am a little unclear about this part of the rules.
Of the three you link, only [two](https://scryfall.com/search?q=is%3Amdfc+ci%3D3&unique=cards&as=grid&order=name) have a color identity of 3, but none of them have 3 colors in any one of their mana cost.

Reply:
yeah I guess there’s no way to test for an mdfc but why is it looking at the main card when it’s not the main card that’s getting cast?

Reply:
it's only looking at the main card if it's outside the Battlefield or the stack, otherwise it's looking at the cast side.

Reply:
Now that I think about it, I wonder if those checks could be refactored into getColor itself.